frontends/ol-components/src/components/Carousel/Carousel.test.tsx

```diff
@@ -1,95 +1,160 @@
 import React from "react"
-import { render, screen } from "@testing-library/react"
-import user from "@testing-library/user-event"
 import { Carousel } from "./Carousel"
-import type { CarouselProps } from "./Carousel"
+import { render, screen, fireEvent, waitFor } from "@testing-library/react"
+import user from "@testing-library/user-event"
 import { ThemeProvider } from "../ThemeProvider/ThemeProvider"
-
-jest.mock("nuka-carousel", () => {
-  const actual = jest.requireActual("nuka-carousel")
-  return {
-    ...actual,
-    default: jest.fn(() => <div />),
-    __esModule: true,
-  }
-})
-
-const setupCarousel = (props?: Partial<CarouselProps>) =>
-  render(
-    <Carousel pageSize={2} {...props}>
-      <div>Child 1</div>
-      <div>Child 2</div>
-      <div>Child 3</div>
-      <div>Child 4</div>
-      <div>Child 5</div>
-    </Carousel>,
-    { wrapper: ThemeProvider },
-  )
-
-const getNextPageButton = () => screen.getByRole("button", { name: "Next" })
-const getPrevPageButton = () => screen.getByRole("button", { name: "Previous" })
+import { faker } from "@faker-js/faker/locale/en"
+
+const mockWidths = ({
```

Comment:
Certainly this test would be better in a real browser.

That said, `slick-carousel`'s class structure and `Carousel.tsx`'s usage of `getBoundingClientRect` is pretty simple, so this was actually pretty easy to handle reasonably with JSDom.